lib/net.c

```diff
@@ -507,28 +579,103 @@ gatekeeper_init_network(struct net_config *net_conf)
 			(net_conf->front.num_ports + net_conf->back.num_ports));
 
 	/* Initialize interfaces. */
-	ret = init_iface(net_conf, &net_conf->front);
+	ret = init_iface(&net_conf->front);
 	if (ret < 0)
 		goto out;
 
-	ret = init_iface(net_conf, &net_conf->back);
+	ret = init_iface(&net_conf->back);
 	if (ret < 0)
-		goto close_front;
+		goto destroy_front;
 
 	goto out;
 
-close_front:
-	close_iface_id(&net_conf->front, net_conf->front.num_ports);
-	close_iface_ports(&net_conf->front, net_conf->front.num_ports);
-	rte_free(net_conf->front.ports);
-	net_conf->front.ports = NULL;
+destroy_front:
+	destroy_iface(&net_conf->front, IFACE_DESTROY_INIT);
 out:
 	return ret;
 }
 
+static int
+start_port(uint8_t port_id, uint8_t *num_succ_ports)
```

Comment:
`num_succ_ports` should be renamed to `pnum_succ_ports` to avoid confusion in the code.